packages/react-router/lib/dom/ssr/routeModules.ts

```diff
@@ -18,19 +18,82 @@ export interface RouteModules {
   [routeId: string]: RouteModule | undefined;
 }
 
+/**
+ * The shape of a route module shipped to the client
+ */
 export interface RouteModule {
+  /**
+   * Action function called only in the browser.  Route client loaders provide
+   * data to route components in addition to, or in place of, route loaders.
+   */
   clientAction?: ClientActionFunction;
+  /**
+   * Like route actions but only called in the browser.
+   */
   clientLoader?: ClientLoaderFunction;
-  ErrorBoundary?: ErrorBoundaryComponent;
-  HydrateFallback?: HydrateFallbackComponent;
+  /**
+   * ErrorBoundary to display for this route
+   */
+  ErrorBoundary?: React.ComponentType;
+  /**
+   * `<Route HydrateFallback>` component to render on initial loads
+   * when client loaders are present
+   */
+  HydrateFallback?: React.ComponentType;
+  /**
+   * The `Layout` export is only applicable to the root route.  Because the root
+   * route manages the document for all routes, it supports an additional optional
+   * `Layout` export that will be wrapped around the rendered UI, which may come
+   * from the default `Component` export, the `ErrorBoundary`, or the `HydrateFallback`
+   */
   Layout?: LayoutComponent;
-  default: RouteComponent;
+  /**
+   * Defines the component that will render when the route matches.
+   */
+  default: React.ComponentType<{}>;
+  /**
+   * Route handle allows apps to add anything to a route match in `useMatches`
+   * to create abstractions (like breadcrumbs, etc.).
+   */
   handle?: RouteHandle;
+  /**
+   * Route links define [`<link>` element][link-element]s to be rendered in the
+   * document `<head>`.
+   */
   links?: LinksFunction;
+  /**
+   * Route meta defines meta tags to be rendered in the `<head>` of the document.
+   */
   meta?: MetaFunction;
+  /**
+   * By default, all routes are revalidated after actions. This function allows
+   * a route to opt-out of revalidation for actions that don't affect its data.
+   */
   shouldRevalidate?: ShouldRevalidateFunction;
 }
 
+/**
+ * The shape of a route module file for your application
+ */
+export interface ServerRouteModule extends RouteModule {
```

Comment:
No need for this to be an `interface`

```suggestion
export type ServerRouteModule = RouteModule & {
```